bin/faucet/src/server/pow.rs

```diff
@@ -34,119 +30,117 @@ const ACTIVE_REQUESTS_TO_INCREASE_DIFFICULTY: usize = REQUESTS_QUEUE_SIZE / MAX_
+
+        if bytes.len() != 48 {
+            // 8 + 8 + 32 bytes
+            return Err(InvalidRequest::MissingPowParameters);
+        }
+
+        let difficulty = u64::from_le_bytes(bytes[0..8].try_into().unwrap()) as usize;
+        let timestamp = u64::from_le_bytes(bytes[8..16].try_into().unwrap());
+        let signature: [u8; 32] = bytes[16..48].try_into().unwrap();
+
+        let challenge = Self::new(difficulty, timestamp, signature);
+
+        // Verify the signature
+        let expected_signature = Self::compute_signature(salt, difficulty, timestamp);
+        if signature != expected_signature {
             return Err(InvalidRequest::ServerSignaturesDoNotMatch);
         }
-        Ok(())
+
+        Ok(challenge)
     }
 
-    /// Check the received timestamp.
-    ///
-    /// The timestamp is valid if it is within `SERVER_TIMESTAMP_TOLERANCE_SECONDS` seconds of the
-    /// current time.
-    pub fn check_server_timestamp(&self) -> Result<(), InvalidRequest> {
-        let server_timestamp = SystemTime::now()
-            .duration_since(UNIX_EPOCH)
-            .expect("Time went backwards")
-            .as_secs();
+    /// Encodes the challenge into a hex string.
+    pub fn encode(&self) -> String {
+        let mut bytes = Vec::with_capacity(48);
+        bytes.extend_from_slice(&(self.difficulty as u64).to_le_bytes());
+        bytes.extend_from_slice(&self.timestamp.to_le_bytes());
+        bytes.extend_from_slice(&self.signature);
+        Self::hex_encode(&bytes)
+    }
 
-        if (server_timestamp - self.server_timestamp) > SERVER_TIMESTAMP_TOLERANCE_SECONDS {
-            return Err(InvalidRequest::ExpiredServerTimestamp(
-                self.server_timestamp,
-                server_timestamp,
-            ));
-        }
+    /// Checks whether the provided nonce satisfies the difficulty requirement encoded in the
+    /// challenge.
+    pub fn validate_pow(&self, nonce: u64) -> bool {
+        let mut hasher = Sha3_256::new();
+        hasher.update(self.encode());
+        hasher.update(nonce.to_string().as_bytes());
+        let hash = hasher.finalize();
+        let hash_hex = hash.to_hex();
 
-        Ok(())
+        let leading_zeros = hash_hex.chars().take_while(|&c| c == '0').count();
+        leading_zeros >= self.difficulty
     }
 
-    /// Check a `PoW` solution.
-    ///
-    /// * `challenge_cache` - The challenge cache to be used to validate the challenge.
-    ///
-    /// The solution is valid if the hash of the seed and the solution has at least `DIFFICULTY`
-    /// leading zeros.
-    pub fn check_pow_solution(
-        &self,
-        challenge_cache: &ChallengeCache,
-    ) -> Result<(), InvalidRequest> {
-        let mut challenges =
-            challenge_cache.challenges.lock().expect("PoW challenge cache lock poisoned");
+    /// Checks if the challenge timestamp is still valid.
+    pub fn is_timestamp_valid(&self) -> bool {
```

Comment:
Not the greatest suggestion either, but I'd like to emphasize that its an expiration check.

```suggestion
    pub fn is_not_expired(&self) -> bool {
```